addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1348,17 +1348,17 @@ def _notify_by_web_push_prepare_payload(self, message, msg_vals=False):
         payload['options']['data']['action'] = 'mail.action_discuss'
         record_name = msg_vals.get('record_name') if msg_vals and 'record_name' in msg_vals else message.record_name
         author_id = [msg_vals.get('author_id')] if 'author_id' in msg_vals else message.author_id.ids
+        author_guest_id = [msg_vals.get('author_guest_id')] if 'author_guest_id' in msg_vals else message.author_guest_id.ids
+        author_name = self.env['res.partner'].browse(author_id).name if author_id[0] else self.env['mail.guest'].browse(author_guest_id).name
```

Comment:
:pick: 
```suggestion
        author = self.env["res.partner"].browse(author_id) or self.env["mail.guest"].browse(
            msg_vals.get("author_guest_id", message.author_guest_id.id)
        )
```

Reply:
We should either use `browse(...).exists()` or `if author_id` IMO. Otherwise we risk `IndexError`s, don't we?

Reply:
change done!